ssl/s23_clnt.c

```diff
@@ -121,22 +121,10 @@ static int ssl23_client_hello(SSL *s);
 static int ssl23_get_server_hello(SSL *s);
 static const SSL_METHOD *ssl23_get_client_method(int ver)
 {
-#ifndef OPENSSL_NO_SSL2
-    if (ver == SSL2_VERSION)
-        return (SSLv2_client_method());
-#endif
-#ifndef OPENSSL_NO_SSL3
-    if (ver == SSL3_VERSION)
-        return (SSLv3_client_method());
-#endif
-    if (ver == TLS1_VERSION)
-        return (TLSv1_client_method());
-    else if (ver == TLS1_1_VERSION)
-        return (TLSv1_1_client_method());
-    else if (ver == TLS1_2_VERSION)
-        return (TLSv1_2_client_method());
-    else
-        return (NULL);
+    /* When SSL_set_session is called, do NOT switch to the version-specific
+     * method table. The server may still negotiate a different version when
+     * rejecting the session. */
+    return SSLv23_client_method();
```

Comment:
Oh yeah, I'm just a bit afraid that if I will make changes way too broad - it will take much longer time to review and land it.